src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -184,7 +177,23 @@ private static Assembly Resolve(IntPtr gchManagedAssemblyLoadContext, AssemblyNa
         {
             AssemblyLoadContext context = (AssemblyLoadContext)(GCHandle.FromIntPtr(gchManagedAssemblyLoadContext).Target);
             
-            return context.LoadFromAssemblyName(assemblyName);
+            // AssemblyName is mutable. Cache the expected name before anybody gets a chance to modify it.
+            string requestedSimpleName = assemblyName.Name;
```

Comment:
I do not think it make sense to pass this as an argument - I think it should be just a local var in `ResolveUsingLoad` / `ResolveUsingResolvingEvent`